### ISSUE TITLE
Feature request: 'config.d' like approach for Guardfiles/templates

### ISSUE THREAD

**Issue by e2** [guard/guard#695]
### TL;DR - each plugin instance could be configured in it's own template file (which would allow plugins to upgrade themselves) and directories could act as groups.

First, Guardfiles (or .Guardfiles) should not be in the main directory (except maybe for a `.guard.rb` file in the project directory with just global settings and no plugins, groups or rules). This is so reloading/watching can work with the `directories` option.

Example (proposed) structure:

```bash
Guardfile # with only Guard config statements like `directories(%w(app lib spec))`, `notification`, `ignores`, etc.

config/backend/group.guardfile # contains group options
config/backend/rspec.guardfile

config/frontend/group.guardfile # contains group options
config/frontend/rspec.guardfile
config/frontend/cucumber.guardfile

config/group.guardfile # options for default group
config/bundler.guardfile
```

Also, since each plugin instance (meaning, in separate folder/group) would have it's own file, it wouldn't have to be declared with `guard`, e.g.:

`config/backend/rspec.guardfile`

```ruby
options cmd: 'bundle exec rspec', failed_mode: :focus

watch(%r{^spec/acceptance/(.+)\.feature$})
# etc.
```

Also, this would eliminate Groups from RSpec (it would just check for folders)

**Comment by ITler** [comment]
I'm looking forward getting this feature into action.
I would like to suggest to think about renaming the sub directory's name `config`. This is too common as it might conflict with folder names from project context. Maybe it is better to give it a more specific name, like rake does with `rakelib`, i.e. `guard.d`.

**Comment by e2** [comment]
@ITler - yes! good idea!

See also: https://github.com/guard/guard/issues/796

**Comment by e2** [comment]
Closing this in favor of: https://github.com/guard/guard/issues/838

(There's a workaround/snippet there you can already use).

I also think that once templates are moved into smaller files, the `Guardfile` stays as the perfect place for defining groups and custom-only watches and custom plugin option overrides.
